File: src/local_education_app/lib/models/tour/tour.dart
// ignore_for_file: public_member_api_docs, sort_constructors_first

class Tour {
  String id;
  String title;
  String urlSlug;
  DateTime createDate;
  int viewCount;
  bool isPublished;
  bool isDeleted;
  String userName;
  String urlPreview;
  Tour({
    required this.id,
    required this.title,
    required this.urlSlug,
    required this.createDate,
    required this.viewCount,
    required this.isPublished,
    required this.isDeleted,
    required this.userName,
    required this.urlPreview,
  });
  factory Tour.fromMap(Map<String, dynamic> json) {
    return Tour(
      id: json['id'],
      title: json['title'],
      urlSlug: json['urlSlug'],
      createDate: DateTime.parse(json['createdDate']),
      viewCount: json['viewCount'],
      isDeleted: json['isPublished'],
      isPublished: json['isDeleted'],
      userName: json['username'],
      urlPreview: json['urlPreview'],
    );
  }
}

File: src/local-education-api/LocalEducation.Core/Entities/Tour.cs
using LocalEducation.Core.Contracts;
using static System.Formats.Asn1.AsnWriter;

namespace LocalEducation.Core.Entities;

public class Tour : IEntity
{
	public Guid Id { get; set; }

	public Guid UserId { get; set; }

	public string Title { get; set; }

	public string UrlSlug { get; set; }

	public DateTime CreatedDate { get; set; }

	public int ViewCount { get; set; }

	public bool IsPublished { get; set; }

	public bool IsDeleted { get; set; }

	// ======================================================
	// Navigation properties
	// ======================================================

	public IList<Scene> Scenes { get; set; }

    public Atlas Atlas { get; set; }

	public User User { get; set; }

	public Tour()
	{
		
	}

	public Tour(Guid id, Guid userId, string title, string urlSlug, DateTime createdDate, int viewCount, bool isPublished, bool isDeleted)
	{
		Id = id;
		UserId = userId;
		Title = title;
		UrlSlug = urlSlug;
		CreatedDate = createdDate;
		ViewCount = viewCount;
		IsPublished = isPublished;
		IsDeleted = isDeleted;
	}
}